sql/country_statistics.sql

```diff
@@ -1,117 +1,140 @@
-          sum(chg.coastline_km_modified + chg.coastline_km_deleted) AS coastline_km_modified,
-          sum(chg.roads_added) AS roads_added,
-          sum(chg.roads_modified + chg.roads_deleted) AS roads_modified,
-          sum(chg.waterways_added) AS waterways_added,
-          sum(chg.waterways_modified + chg.waterways_deleted) AS waterways_modified,
-          sum(chg.coastlines_added) AS coastlines_added,
-          sum(chg.coastlines_modified + chg.coastlines_deleted) AS coastlines_modified,
-          sum(chg.buildings_added) AS buildings_added,
-          sum(chg.buildings_modified + chg.buildings_deleted) AS buildings_modified,
-          sum(chg.pois_added) AS pois_added,
-          sum(chg.pois_modified + chg.pois_deleted) AS pois_modified,
-          max(coalesce(chg.closed_at, chg.created_at)) AS last_edit,
-          max(COALESCE(chg.closed_at, chg.created_at, chg.updated_at)) AS updated_at,
-          count(*) AS changeset_count,
-          sum(cc.edit_count) AS edit_count
-        FROM (changesets chg
-          JOIN country_counts cc ON ((cc.changeset_id = chg.id)))
-        GROUP BY cc.id
-    )
-  SELECT agg.country_id,
-      countries.name AS country_name,
-      countries.code AS country_code,
-      agg.road_km_added,
-      agg.road_km_modified,
-      agg.waterway_km_added,
-      agg.waterway_km_modified,
-      agg.coastline_km_added,
-      agg.coastline_km_modified,
-      agg.roads_added,
-      agg.roads_modified,
-      agg.waterways_added,
-      agg.waterways_modified,
-      agg.coastlines_added,
-      agg.coastlines_modified,
-      agg.buildings_added,
-      agg.buildings_modified,
-      agg.pois_added,
-      agg.pois_modified,
-      agg.last_edit,
-      agg.updated_at,
-      agg.changeset_count,
-      agg.edit_count,
-      jce.edits AS user_edit_counts,
-      hts.hashtag_edits
-    FROM (agg_stats agg
-      FULL OUTER JOIN json_country_edits jce ON (agg.country_id = jce.country_id)
-      FULL OUTER JOIN ht_json hts ON (agg.country_id = hts.country_id)
-      JOIN countries ON agg.country_id = countries.id);
+  WITH changesets AS (
+    SELECT
+      *
+    FROM changesets
+    -- ignore users 0 and 1
+    WHERE user_id > 1
+  ),
+  general AS (
+    SELECT
+      country_id,
+      max(coalesce(closed_at, created_at)) last_edit,
+      count(*) changeset_count,
+      sum(edit_count) edit_count,
+      max(updated_at) updated_at
+    FROM changesets
+    JOIN changesets_countries ON changesets.id = changesets_countries.changeset_id
+    GROUP BY country_id
+  ),
+  processed_changesets AS (
+    SELECT
+      -- TODO include changesets_countries.edit_count as an alternative to changeset count
+      id,
+      user_id,
+      country_id,
+      measurements,
+      counts
+    FROM changesets
+    JOIN changesets_countries ON changesets.id = changesets_countries.changeset_id
+  ),
+  hashtag_counts AS (
+    SELECT
+      -- TODO rank by edit count?
```

Comment:
@giblet22 this produces the top 10 hashtags by number of changesets (and surfaces that count in the response). Is that the correct thing to sort by? (It could also make sense to sort by number of edits.)

Reply:
Will confirm with customer, but I agree top hashtags by edits would make sense for a better picture overall.

Reply:
👍